source/javascripts/hooks/useEnvVars.ts

```diff
@@ -0,0 +1,159 @@
+import { useMemo } from 'react';
+import { useShallow } from 'zustand/react/shallow';
+import { useQueries, useQuery } from '@tanstack/react-query';
+import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
+import StepService from '@/core/models/StepService';
+import { EnvVar, EnvVarYml } from '@/core/models/EnvVar';
+import EnvVarService from '@/core/models/EnvVarService';
+import { WorkflowYmlObject } from '@/core/models/Workflow';
+import EnvVarsApi from '@/core/api/EnvVarsApi';
+import WindowUtils from '@/core/utils/WindowUtils';
+import StepApi from '@/core/api/StepApi';
+
+function envVarsAreEqual(a: EnvVar, b: EnvVar) {
+  return a.key === b.key && a.source === b.source;
+}
+
+const useCvsFromYml = (workflowId: string) => {
+  return useBitriseYmlStore(
+    useShallow((s) => {
+      const cvss: string[] = [];
+
+      const collectStepCvssOf = (workflow?: WorkflowYmlObject) => {
+        workflow?.steps?.forEach((ymlStepObject) => {
+          const [cvs, step] = Object.entries(ymlStepObject)[0];
+
+          // TODO: Handle step bundles and with groups...
+          if (!StepService.isStepBundle(cvs, step) && !StepService.isWithGroup(cvs, step) && !cvss.includes(cvs)) {
+            cvss.push(cvs);
+          }
+        });
+      };
+
+      collectStepCvssOf(s.yml.workflows?.[workflowId]);
+
+      s.yml.workflows?.[workflowId]?.before_run?.forEach((chainWorkflowId) => {
+        collectStepCvssOf(s.yml.workflows?.[chainWorkflowId]);
+      });
+
+      s.yml.workflows?.[workflowId]?.after_run?.forEach((chainWorkflowId) => {
+        collectStepCvssOf(s.yml.workflows?.[chainWorkflowId]);
+      });
+
+      return cvss;
+    }),
+  );
+};
+
+const useEnvVarsFromYml = (workflowId: string) => {
+  return useBitriseYmlStore(
+    useShallow((s) => {
+      const envs: EnvVar[] = [];
+
+      const addEnvVar = (env: EnvVarYml, source?: string) => {
+        const envVar = EnvVarService.parseYmlEnvVar(env, source);
+        if (!envs.some((e) => envVarsAreEqual(e, envVar))) {
+          envs.push(envVar);
+        }
+      };
+
+      s.yml.app?.envs?.forEach((env) => {
```

Comment:
I am not sure I understand the importance the "merge order" here. Since workflow env vars can override app env vars (right?) they should be merged at the end?